core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -448,6 +446,47 @@ static HttpRequestInterceptor loadInterceptorDynamically(
     return instance;
   }
 
+  static HttpClientConnectionManager configureConnectionManager(
+      Map<String, String> properties, ConnectionConfig connectionConfig) {
```

Comment:
rather than passing in the `ConnectionConfig` I would probably just call `configureConnectionConfig(properties)` inside `configureConnectionManager(..)`